Q2/2022 Roadmap (#194)
meilisearch/integration-guides: roadmap/2022.md

  - [ ] Finish changes regarding the API keys
  - [ ] Finish changes regarding the new index routes
  - [ ] Add the tenant tokens methods

## Q2

Analytics:
- [ ] Finish the implementation in the remaining SDKs (Java, Javascript, Rust)
- [ ] Plan how we will implement the analytics on the integrations, tools & plugins.

> - [ ] _(may not be required)_ Define the specs to support analytics in some tools with core team.

Reviewer: ```suggestion
- [ ] _(may not be required)_ Define the specs to support analytics in front-end tools with core team.
```

> - [ ] Implement it on the integrations.
> - [ ] Implement it on the plugins.

Reviewer: ```suggestion
- [ ] Implement it on the all the integrations maintained by the integration team.
```

Reviewer: I'm not sure what you mean by "plugin" regarding [this](https://github.com/meilisearch/integration-guides#-our-integrations)
You need to implement it in
- web framework integration
- front-end integration
- platform plugin
- devOps tools (kubernetes only)
- the other integrations without any category: docs-scaper

Author: Oh yeah, I meant "platform plugins". But I like the non-sdk suggestion from @bidoubiwa.

But to be honest, I put that in different items to have a sense of progress and at the same time, I don't want to specify all of the items individually. 
So I'll accept your suggestion :)

Reviewer: If we have not finished all the implementation by the end of the quarter, we could add the list of the already finished ones :)

Processes:

> - [ ] Change the Integrations' Team developers scope.

Reviewer: ```suggestion
- [ ] Change the Integrations' Team developers scope: make it adapted to each developer, but also to reduce the context switches.
```

Improvements:
- [ ] Add a helpful docker/compose setup in every integration.
- [ ] Code coverage:
  - [ ] Define the tool which will be used (codecov, coveralls, others).
  - [ ] Implement it in the SDKs.
  - [ ] Implement it on the integrations which makes sense to have coverage.

> Meilisearch releases:

Reviewer: Don't forget v0.28.0. Even if you cannot add a lot of details, you can say the routes will be stabilized (regarding naming and behavior) and will involve a lot of work for the integration team

Author: I've added some items about it with a total of 43% of confidence 😆 ! Check it out, please :)

  - [ ] Finish changes regarding the API keys
  - [ ] Finish changes regarding the new index routes
  - [ ] Add the tenant tokens methods

## Q2

> 

Reviewer: ```suggestion
```

Analytics:
- [ ] Finish the implementation in the remaining SDKs (Java, Javascript, Rust)

> - [ ] Plan how we will implement the analytics on the integrations, tools & plugins.

Reviewer 2: > integrations, tools & plugins

It would be more easy if we had a word for `non-sdk`, just saying hehehe

- [ ] _(may not be required)_ Define the specs to support analytics in some tools with core team.
- [ ] Implement it on the integrations.
- [ ] Implement it on the plugins.

Processes:
- [ ] Change the Integrations' Team developers scope.

Improvements:
- [ ] Add a helpful docker/compose setup in every integration.
- [ ] Code coverage:
  - [ ] Define the tool which will be used (codecov, coveralls, others).
  - [ ] Implement it in the SDKs.
  - [ ] Implement it on the integrations which makes sense to have coverage.

Meilisearch releases:

> - [ ] v0.27 work

Reviewer 2: Should we add v0.28 ?

Author: Yeah, I think, but currently, I don't have the time to dig into the planned changes for the v0.28.

Author: And I don't want to put "v0.28 a lot of things" 😅

- [ ] Plan how we will implement the analytics on the non-sdk.
- [ ] _(may not be required)_ Define the specs to support analytics in front-end tools with core team.
- [ ] Implement it on the all the integrations maintained by the integration team.

Processes:
- [ ] Change the Integrations' Team developers scope: make it adapted to each developer, but also to reduce the context switches.

Improvements:
- [ ] Add a helpful docker/compose setup in every integration.
- [ ] Code coverage:
  - [ ] Define the tool which will be used (codecov, coveralls, others).
  - [ ] Implement it in the SDKs.
  - [ ] Implement it on the integrations which makes sense to have coverage.

Meilisearch releases:
- [ ] v0.27 work:
  - [ ] Docker run changes
  - [ ] Support nested fields
  - [ ] Customize the typo tolerance
  - [ ] Add support to use the new cropping and highlighting search attributes
- [ ] v0.28 work _(to be completed when the final pre-release comes)_:
  - [ ] New API namings, could impact multiple resources and nomenclatures, like dumps, keys, search, and indexes.
  - [ ] New behavior of the keys expiresAt field.

>   - [ ] Improve understanding of the pagination of search results.

Reviewer: Not on the integration team side (as far as I know). Mostly a work for docs team

>   - [ ] Add pagination to the tasks API.

Reviewer: ```suggestion
  - [ ] Add pagination to the multiple API resources (keys, tasks, indexes, documents).
```

>   - [ ] Improve API keys creation to be made in a deterministic way.

Reviewer: Not sure it will impact the integration team
